Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/common/DccTicketingException.kt

```diff
@@ -0,0 +1,96 @@
+package de.rki.coronawarnapp.dccticketing.core.common
+
+import android.content.Context
+import androidx.annotation.StringRes
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.util.HasHumanReadableError
+import de.rki.coronawarnapp.util.HumanReadableError
+import de.rki.coronawarnapp.util.ui.CachedString
+import de.rki.coronawarnapp.util.ui.LazyString
+
+class DccTicketingException(
+    val errorCode: ErrorCode,
+    override val cause: Throwable? = null
+) : Exception(errorCode.message, cause), HasHumanReadableError {
+
+    @Suppress("MaxLineLength")
+    enum class ErrorCode(val message: String) {
+        AES_CBC_INVALID_IV("failure while encrypting DCC; user may retry"),
+        AES_CBC_INVALID_KEY("failure while encrypting DCC; user may retry"),
+        AES_CBC_NOT_SUPPORTED("failure while encrypting DCC; user may retry"),
+        AES_GCM_INVALID_IV("failure while encrypting DCC; user may retry"),
+        AES_GCM_INVALID_KEY("failure while encrypting DCC; user may retry"),
+        AES_GCM_NOT_SUPPORTED("failure while encrypting DCC; user may retry"),
+
+        ATR_AUD_INVALID("failure when obtaining Access Token; user may retry"),
+        ATR_CERT_PIN_MISMATCH("failure when obtaining Access Token; user may retry"),
+        ATR_CERT_PIN_NO_JWK_FOR_KID("failure when obtaining Access Token; user may retry"),
+        ATR_CLIENT_ERR("failure when obtaining Access Token; user may retry"),
+        ATR_JWT_VER_ALG_NOT_SUPPORTED("failure when obtaining Access Token; user may retry"),
+        ATR_JWT_VER_EMPTY_JWKS("failure when obtaining Access Token; user may retry"),
+        ATR_JWT_VER_NO_JWK_FOR_KID("failure when obtaining Access Token; user may retry"),
+        ATR_JWT_VER_NO_KID("failure when obtaining Access Token; user may retry"),
+        ATR_JWT_VER_SIG_INVALID("failure when obtaining Access Token; user may retry"),
+        ATR_NO_NETWORK("failure when obtaining Access Token; user may retry"),
+        ATR_PARSE_ERR("failure when obtaining Access Token; user may retry"),
+        ATR_SERVER_ERR("failure when obtaining Access Token; user may retry"),
+        ATR_TYPE_INVALID("failure when obtaining Access Token; user may retry"),
+
+        EC_SIGN_INVALID_KEY("failure while encrypting DCC; user may retry"),
+        EC_SIGN_NOT_SUPPORTED("failure while encrypting DCC; user may retry"),
+
+        INIT_DATA_PARSE_ERR("invalid QR code"),
```

Comment:
Can you remove unrelated error codes? They are covered in other parts, eg. the extractor

Reply:
I think we should differentiate between errors that occur when parsing the qr code, errors from API calls and errors from internal processing, eg key generation or decryption. The prefix already groups the codes.

Reply:
might be cleaned up later, but no urgent yet I guess..